### PR TITLE
functionality for filter products added

### PR DESCRIPTION
**Pull request by aishwarya-bm**
functionality for filter products added - filter by various options, clear filters
also used reducer, context to manage state for filters

preview at:

https://user-images.githubusercontent.com/66695068/160207920-d03473ea-4c40-495b-ac34-53e12be801a0.mp4

### CODE REVIEW

File: src/components/filters/FilterBy/PriceFilter.jsx [hunk 1]
```diff
@@ -9,15 +12,15 @@ export default function PriceFilter() {
             type="range"
             min="1000"
```
Comment (therajatg) [inline src/components/filters/FilterBy/PriceFilter.jsx:13]: I saw that u have many products below 1000. So, u may want to start from a lower number.

File: src/components/filters/FilterBy/PriceFilter.jsx [hunk 2]
```diff
@@ -9,15 +12,15 @@ export default function PriceFilter() {
...
             max="5000"
-            // value={state.priceLimit}
+            value={stateFilter.priceLimit}
             list="tickmarks"
             step="1000"
```
Comment (therajatg) [inline src/components/filters/FilterBy/PriceFilter.jsx:17]: I saw that u have many products below 1000. So, u may want to start from a lower number.

**Review by therajatg — APPROVED** [review]

File: src/components/filters/FilterBy/PriceFilter.jsx (hunk 1 above)
Comment (therajatg) [inline src/components/filters/FilterBy/PriceFilter.jsx]: I see that you have products below 1000 too. So, u may wanna start range from a lower number.

Comment (aishwarya-bm) [reply src/components/filters/FilterBy/PriceFilter.jsx]: Made the changes, thanks!

**Review by sadathshariff — APPROVED** [review]
The Code Looks Clean and liked how you have broken down the Filter Component

**Pull request merged**